tools/train.py

```diff
@@ -77,8 +77,10 @@ def main():
         cfg.work_dir = args.work_dir
     elif cfg.get('work_dir', None) is None:
         # use config filename as default work_dir if cfg.work_dir is None
-        cfg.work_dir = osp.join('./work_dirs',
-                                osp.splitext(osp.basename(args.config))[0])
+        config_dir = osp.split(osp.dirname(args.config))[1]
+        config_name = osp.splitext(osp.basename(args.config))[0]
+        cfg.work_dir = osp.join('./work_dirs', config_dir, config_name)
```

Comment:
Please take the following case into consideration:

The user added 1 config file under the folder `configs/`, or directory under the root of repo. `config_dir` will become `configs` or `MMSegmentation`, which is not idea.

Reply:
I suggest keep this part unchanged.